optuna/_gp/acqf.py

```diff
@@ -22,6 +24,41 @@
     torch = _LazyImport("torch")
 
 
+def _sample_from_normal_sobol(dim: int, n_samples: int, seed: int | None) -> torch.Tensor:
```

Comment:
Uniform sampling based on Sobol sequence and the inverse cumulative distribution function is a good method with fast convergence, especially effective in low-dimensional problems. However, compared to simply sampling normally distributed random variables using methods like `torch.randn`, is this approach really suitable for the current purpose? Are there any references or experimental results that support the use of this method?

Reply:
```
        # NOTE(nabenabe): By using fixed samples from the Sobol sequence, EHVI becomes
        # deterministic, making it possible to optimize the acqf by l-BFGS.
```

This comment helped me understand the necessity of this function. Since the SobolEngine is re-initialized with the same random seed each time, it makes sense that exactly the same points are sampled every time a new instance of MultiObjectiveAcquisitionFunctionParams is created. On the other hand, I'm curious whether it's better to use this approach or to use torch.randn with a newly initialized generator as shown below. What do you think?

```python
g = torch.Generator()
g.manual_seed(seed)
samples = torch.randn(n_samples, dim, dtype=torch.float64, generator=g)
```

Reply:
The reference is available at Appendix D of [this paper](https://arxiv.org/pdf/2006.05078), but basically, the authors checked four scenarios:

![image](https://github.com/user-attachments/assets/d42bb231-8a3c-4045-8ae2-3ed6dfa898e0)

Reply:
I understood. Sounds very good to use QMC fixed samples.